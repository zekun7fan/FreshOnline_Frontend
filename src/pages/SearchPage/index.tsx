import React, {useState} from "react";
import {Layout, Breadcrumb} from 'antd';

import SearchHeaderUI from "../../components/SearchHeader";
import SearchSiderUI from "../../components/SearchSider";
import SearchContentUI from "../../components/SearchContent";
import {Link} from "react-router-dom";

const { Header, Content, Sider, Footer } = Layout;

export default function SearchPage() {
    return (
        <Layout>
            <Layout>
                <Sider width={200} className="site-layout-background">
                    <SearchSiderUI />
                </Sider>
                <Layout style={{ padding: '0 24px 24px' }}>
                    <Breadcrumb style={{ margin: '16px 0' }}>
                        <Breadcrumb.Item><Link to='/'>Home</Link></Breadcrumb.Item>
                        <Breadcrumb.Item>Search</Breadcrumb.Item>
                    </Breadcrumb>
                    <Content className="site-layout-background"
                        style={{
                            padding: 24,
                            margin: 0,
                            minHeight: 280,
                        }}>
                        <SearchContentUI/>
                    </Content>
                </Layout>
            </Layout>
            <SearchFooter/>
        </Layout>
    );
}

export function SearchFooter() {
    return (
        <Layout>
            <Footer style={{ textAlign: 'center' }}>FreshOnline Â©2022 Created by Team FreshOnline</Footer>
        </Layout>
    )
}