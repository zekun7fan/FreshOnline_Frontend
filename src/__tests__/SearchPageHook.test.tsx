import {renderHook, act, cleanup} from '@testing-library/react-hooks';

import SearchPage, {SearchFooter} from "../pages/SearchPage";
import {Provider} from "react-redux";
import store from "../redux/store";
import {BrowserRouter} from "react-router-dom";

jest.mock('antd/es/tree-select', () => ({}));

test('render search page UI using hook', () => {
    const wrapper = ({children}: { children: any }) => (
        <Provider store={store}>
            <BrowserRouter>
                {children}
            </BrowserRouter>
        </Provider>
    )
    const { result } = renderHook(() => SearchPage(), { wrapper })
    expect(result.current).toMatchSnapshot()
})

test('render search footer UI using hook', () => {
    const wrapper = ({children}: { children: any }) => (
        <Provider store={store}>
            <BrowserRouter>
                {children}
            </BrowserRouter>
        </Provider>
    )
    const { result } = renderHook(() => SearchFooter(), { wrapper })
    expect(result.current).toMatchSnapshot()
})