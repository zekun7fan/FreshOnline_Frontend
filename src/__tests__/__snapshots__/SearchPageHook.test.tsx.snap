// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render search footer UI using hook 1`] = `
<Layout>
  <Divider />
  <Footer
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    FreshOnline Â©2022 Created by Team FreshOnline
  </Footer>
</Layout>
`;

exports[`render search page UI using hook 1`] = `
<Layout>
  <Layout>
    <Sider
      className="site-layout-background"
      width={200}
    >
      <SearchSiderUI />
    </Sider>
    <Layout
      style={
        Object {
          "padding": "0 24px 24px",
        }
      }
    >
      <Breadcrumb
        style={
          Object {
            "margin": "16px 0",
          }
        }
      >
        <BreadcrumbItem>
          <Link
            to="/"
          >
            Home
          </Link>
        </BreadcrumbItem>
        <BreadcrumbItem>
          Search
        </BreadcrumbItem>
      </Breadcrumb>
      <Content
        className="site-layout-background"
        style={
          Object {
            "margin": 0,
            "minHeight": 280,
            "padding": 24,
          }
        }
      >
        <SearchContentUI />
      </Content>
    </Layout>
  </Layout>
</Layout>
`;
